docs/activity.md

```diff
@@ -0,0 +1,241 @@
+
+
+## Reacting to events
+
+You'll see at the top of `main.py` a function called `hello()` with a decorator
+`@bird.react("hello")` on top:
+
+```python
+@bird.react("hello")
+def hello():
+    """Somebody is saying hi, how lovely!"""
+    display.show(Image.HEART)
+    sleep(400)
+```
+
+When the bird detects a hello message from the airwaves, it will run this
+function and show a heart image in the micro:bit display for 400 milliseconds.
+
+<p class="exercise">
+Exercise: <br>
+Birds can be very chatty, so your neighbours might be saying hello quite often.
+<br>
+Can you think of a way to alternate between showing `Image.HEART` and
+`Image.HEART_SMALL`? That way it'll look like a beating heart!
+</p>
+
+[Solution](https://github.com/microbit-carlos/microbit-bird-activity/tree/main/solutions/1-hello-beating.md)
+
+When you code is ready, click the purple "Send to micro:bit" button.
+If WebUSB is not available in your browser, you can click the "Save" button
+to download a hex file, and then copy that file into the `MICROBIT` USB drive.
+
+### What was that! A cat?
+
+There are several events the bird can react to, and seeing a cat is quite important for bird survival.
+
+<p class="exercise">
+Exercise: <br>
+Let's do the following:<br>
+- Copy and paste the "hello" example <br>
+- Replace the function name to `cat()` <br>a
+- Change the decorator argument from "hello" to "cat".<br>
+<br>
+What do you think the bird should do in this case?
+<br>
+Some examples of thing you can do can be found in the
+<a href="https://microbit-carlos.github.io/microbit-bird-activity/micropython/#quick-things-to-do-with-the-microbit" target="_blank">MicroPython page</a>.
+</p>
+
+We also want to warn other birds around us,
+let's do that by adding by this line at the end of the function :
+
+```python
+bird.warn_about_cat()
+```
+
+[Solution](https://github.com/microbit-carlos/microbit-bird-activity/tree/main/solutions/2-cat.md)
+
+
+## Bird state
+
+We've learned how to react to environment events (received via micro:bit
+radio), what about things that happen directory to our bird?
+
+Let's look further down `main.py` and we'll find this infinite loop.
+
+```python
+while True:
+    # Look inside yourself and listen, how is your bird feeling?
+    bird_state = bird.current_state()
+    if bird_state == "chill":
+        display.show(Image.HAPPY)
+    sleep(100)
+```
+
+The function `bird.current_state()` uses the micro:bit on board sensors to
+figure how your bird is doing.
+
+<p class="exercise">
+Exercise: <br>
+Birds don't like to be shaken, so "angry" can also be one of the bird states
+to code.
+<br>
+Let's expand that "if" statement with an "elif" and do something else when
+the bird is "angry".
+</p>
+
+[Solution](https://github.com/microbit-carlos/microbit-bird-activity/tree/main/solutions/3-state-angry.md)
+
+
+## SoundEffects
+
+TODO: Need to explain this, and make it appealing for users to spend most of
+the time here.
```

Comment:
```suggestion
```suggestion
But hold on. What's with all these images on screens? Aside from their uncanny ability to fly (sorry Kiwi, Ostrich, Kakapo, Penguin, etc :( ) birds are probably most famous for the sounds they make

The micro:bit V2 has a built-in speaker, and a lot of the fun of building birds is giving them a unique voice.
```
```

Reply:
This is perfect, thanks!